netket/operator/_sumoperators.py

```diff
@@ -40,6 +40,21 @@ def __init__(
             coefficients: A coefficient for each ContinuousOperator object
             dtype: Data type of the matrix elements. Defaults to `np.float64`
         """
+        if len(operators) != len(coefficients):
+            raise AssertionError("Each operator needs a coefficient")
+
```

Comment:
It's a minor convention, but we keep the validation of inputs at the top of the function, so this block should go above here.

```python
        hil = [op.hilbert for op in operators]
          if not all(_ == hil[0] for _ in hil):
            raise NotImplementedError(
                "Cannot add operators on different hilbert spaces"
            )
```